Error connecting to Kafka with SCRAM-SHA-512 mechanism when running as native

### Expected Behavior

The issue looks like https://github.com/micronaut-projects/micronaut-kafka/issues/264 but with another connection options. 
Successful connection is expected.

### Actual Behaviour

```
org.apache.kafka.common.errors.SaslAuthenticationException: Failed to configure SaslClientAuthenticator
Caused by: org.apache.kafka.common.errors.SaslAuthenticationException: Failed to create SaslClient with mechanism SCRAM-SHA-512
15:28:33.816 [kafka-producer-network-thread | producer-2] INFO  o.a.k.c.network.SaslChannelBuilder - [Producer clientId=producer-2] Failed to create channel due to 
org.apache.kafka.common.errors.SaslAuthenticationException: Failed to configure SaslClientAuthenticator
Caused by: org.apache.kafka.common.errors.SaslAuthenticationException: Failed to create SaslClient with mechanism SCRAM-SHA-512
15:28:33.816 [kafka-producer-network-thread | producer-2] WARN  o.apache.kafka.clients.NetworkClient - [Producer clientId=producer-2] Error connecting to node a3f29642800cd4fc488efa01ad323788-1018565300.eu-west-1.elb.amazonaws.com:9094 (id: -1 rack: null)
```

After I enabled debug in kafka library, the latest successful log was:

15:28:33.816 [pool-1-thread-2] DEBUG o.a.k.c.s.a.SaslClientAuthenticator - [Consumer clientId=<snip>, groupId=<snip>] Creating SaslClient: client=null;service=kafka;serviceHostname=<snip>.eu-west-1.elb.amazonaws.com;mechs=[SCRAM-SHA-512]

I believe, we received null here https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslClientAuthenticator.java#L221 because Sasl library was unable to find mechanism requested.

### Steps To Reproduce

Default micronaut-kafka example, compiled natively with following settings. Can be verified without actual kafka server setup, the error occurs before connection.

```
kafka:
  bootstrap.servers: [server]
  security.protocol: SASL_PLAINTEXT
  sasl.jaas.config: org.apache.kafka.common.security.scram.ScramLoginModule required username="[my-key]" password="[my-password]";
  sasl.mechanism: SCRAM-SHA-512
```

### Environment Information

OpenJDK11

### Example Application

_No response_

### Version

3.2.3

--- a/.gitattributes
+++ b/.gitattributes
@@ -1,18 +1,19 @@
 # Auto detect text files and perform LF normalization
 *        text=auto
 
-*.java   text
-*.html   text
-*.kt     text
-*.kts    text
-*.md     text diff=markdown
+*.java   text eol=lf
+*.groovy text eol=lf
+*.html   text eol=lf
+*.kt     text eol=lf
+*.kts    text eol=lf
+*.md     text diff=markdown eol=lf
 *.py     text diff=python executable
 *.pl     text diff=perl executable
 *.pm     text diff=perl
-*.css    text diff=css
-*.js     text
-*.sql    text
-*.q      text
+*.css    text diff=css eol=lf
+*.js     text eol=lf
+*.sql    text eol=lf
+*.q      text eol=lf
 
 *.sh     text eol=lf
 gradlew  text eol=lf

--- a/kafka/src/main/resources/META-INF/native-image/io.micronaut.kafka/micronaut-kafka/native-image.properties
+++ b/kafka/src/main/resources/META-INF/native-image/io.micronaut.kafka/micronaut-kafka/native-image.properties
@@ -15,5 +15,5 @@
 #
 
 Args = --initialize-at-build-time=org.apache.kafka,net.jpountz,com.sun.org.slf4j.internal.Logger \
-       -H:AdditionalSecurityProviders=com.sun.security.sasl.Provider \
+       -H:AdditionalSecurityProviders=com.sun.security.sasl.Provider,org.apache.kafka.common.security.scram.internals.ScramSaslClientProvider,org.apache.kafka.common.security.oauthbearer.internals.OAuthBearerSaslClientProvider \
        --initialize-at-run-time=org.apache.kafka.common.security.authenticator.SaslClientAuthenticator,io.micronaut.configuration.kafka.health.$KafkaHealthIndicator$Definition,io.micronaut.configuration.kafka.tracing.$KafkaConsumerTracingInstrumentation$Definition,io.micronaut.configuration.kafka.tracing.$KafkaProducerTracingInstrumentation$Definition,io.micronaut.configuration.kafka.tracing.brave.$BraveKafkaConsumerTracingInstrumentation$Definition,io.micronaut.configuration.kafka.tracing.brave.$BraveKafkaProducerTracingInstrumentation$Definition,io.micronaut.configuration.kafka.tracing.brave.$BraveKafkaTracingFactory$KafkaTracing0$Definition

--- a/kafka/src/main/resources/META-INF/native-image/io.micronaut.kafka/micronaut-kafka/reflect-config.json
+++ b/kafka/src/main/resources/META-INF/native-image/io.micronaut.kafka/micronaut-kafka/reflect-config.json
@@ -20,7 +20,17 @@
     "allDeclaredConstructors":true
   },
   {
+    "name": "org.apache.kafka.common.security.scram.ScramSaslClient",
+    "allDeclaredConstructors": true,
+    "allDeclaredMethods": true
+  },
+  {
     "name":"org.apache.kafka.common.security.authenticator.DefaultLogin",
+    "allDeclaredMethods":true,
+    "allDeclaredConstructors":true
+  },
+  {
+    "name":"org.apache.kafka.common.security.plain.PlainLoginModule",
     "allDeclaredMethods":true,
     "allDeclaredConstructors":true
   },
@@ -30,7 +40,32 @@
     "allDeclaredConstructors":true
   },
   {
-    "name":"org.apache.kafka.common.security.plain.PlainLoginModule",
+    "name": "org.apache.kafka.common.security.scram.ScramLoginModule",
+    "allDeclaredConstructors": true,
+    "allDeclaredMethods": true
+  },
+  {
+    "name": "org.apache.kafka.common.security.scram.internals.ScramSaslClient$ScramSaslClientFactory",
+    "allDeclaredConstructors": true,
+    "allDeclaredMethods": true
+  },
+  {
+    "name": "org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule",
+    "allDeclaredConstructors": true,
+    "allDeclaredMethods": true
+  },
+  {
+    "name": "org.apache.kafka.common.security.oauthbearer.internals.OAuthBearerSaslClient",
+    "allDeclaredConstructors": true,
+    "allDeclaredMethods": true
+  },
+  {
+    "name": "org.apache.kafka.common.security.oauthbearer.internals.OAuthBearerSaslClient$OAuthBearerSaslClientFactory",
+    "allDeclaredConstructors": true,
+    "allDeclaredMethods": true
+  },
+  {
+    "name":"org.apache.kafka.common.security.oauthbearer.internals.OAuthBearerSaslClientCallbackHandler",
     "allDeclaredMethods":true,
     "allDeclaredConstructors":true
   },

--- a/kafka/src/test/groovy/io/micronaut/configuration/kafka/AbstractKafkaContainerSpec.groovy
+++ b/kafka/src/test/groovy/io/micronaut/configuration/kafka/AbstractKafkaContainerSpec.groovy
@@ -11,7 +11,7 @@ import spock.lang.Shared
 
 abstract class AbstractKafkaContainerSpec extends AbstractKafkaSpec {
 
-    @Shared @AutoCleanup KafkaContainer kafkaContainer = new KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka:7.0.4")).withEnv(getEnvVariables())
+    @Shared @AutoCleanup KafkaContainer kafkaContainer = new KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka")).withEnv(getEnvVariables())
     @Shared @AutoCleanup ApplicationContext context
 
     void setupSpec() {

--- a/kafka/src/test/groovy/io/micronaut/configuration/kafka/annotation/KafkaProducerSpec.groovy
+++ b/kafka/src/test/groovy/io/micronaut/configuration/kafka/annotation/KafkaProducerSpec.groovy
@@ -19,6 +19,7 @@ import org.apache.kafka.common.header.internals.RecordHeaders
 import org.apache.kafka.common.serialization.StringDeserializer
 import org.apache.kafka.common.serialization.StringSerializer
 import org.testcontainers.containers.KafkaContainer
+import org.testcontainers.utility.DockerImageName
 import spock.lang.AutoCleanup
 import spock.lang.Shared
 
@@ -35,7 +36,7 @@ class KafkaProducerSpec extends AbstractKafkaSpec {
     public static final String TOPIC_QUANTITY = "KafkaProducerSpec-users-quantity"
 
     @Shared MockTracer mockTracer = new MockTracer()
-    @Shared @AutoCleanup KafkaContainer kafkaContainer = new KafkaContainer()
+    @Shared @AutoCleanup KafkaContainer kafkaContainer = new KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka"))
     @Shared @AutoCleanup ApplicationContext context
 
     void setupSpec() {

--- a/kafka/src/test/groovy/io/micronaut/configuration/kafka/docs/DocTests.java
+++ b/kafka/src/test/groovy/io/micronaut/configuration/kafka/docs/DocTests.java
@@ -12,7 +12,7 @@ import org.testcontainers.utility.DockerImageName;
 public class DocTests {
 
     static ApplicationContext applicationContext;
-    static KafkaContainer kafkaContainer = new KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka:7.0.4"));
+    static KafkaContainer kafkaContainer = new KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka"));
 
     @BeforeClass
     public static void setup() {

--- a/kafka/src/test/groovy/io/micronaut/configuration/kafka/health/KafkaHealthIndicatorSpec.groovy
+++ b/kafka/src/test/groovy/io/micronaut/configuration/kafka/health/KafkaHealthIndicatorSpec.groovy
@@ -20,7 +20,7 @@ class KafkaHealthIndicatorSpec extends Specification {
 
     void "test kafka health indicator - UP"() {
         given:
-        KafkaContainer kafkaContainer = new KafkaContainer()
+        KafkaContainer kafkaContainer = new KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka"))
         kafkaContainer.start()
         ApplicationContext applicationContext = ApplicationContext.run(
                 "kafka.bootstrap.servers": kafkaContainer.bootstrapServers
@@ -61,7 +61,7 @@ class KafkaHealthIndicatorSpec extends Specification {
     @Unroll
     void "test kafka health indicator - disabled (#configvalue)"() {
         given:
-        KafkaContainer container = new KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka:7.0.4"))
+        KafkaContainer container = new KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka"))
         container.start()
         ApplicationContext applicationContext = ApplicationContext.run(
                 (AbstractKafkaConfiguration.DEFAULT_BOOTSTRAP_SERVERS): container.bootstrapServers,

--- a/tests/tasks-sasl-plaintext/src/test/groovy/example/TasksSpec.groovy
+++ b/tests/tasks-sasl-plaintext/src/test/groovy/example/TasksSpec.groovy
@@ -18,7 +18,7 @@ class TasksSpec extends Specification implements TestPropertyProvider {
 
     @Shared
     @AutoCleanup
-    KafkaContainer kafkaContainer = new KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka:7.0.4"))
+    KafkaContainer kafkaContainer = new KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka"))
             .withEnv("KAFKA_LISTENER_SECURITY_PROTOCOL_MAP", "BROKER:PLAINTEXT,PLAINTEXT:SASL_PLAINTEXT")
             .withEnv("KAFKA_SASL_ENABLED_MECHANISMS", "PLAIN")
             .withEnv("KAFKA_SASL_MECHANISM_INTER_BROKER_PROTOCOL", "PLAIN")

